sdk/identity/identity/src/credentials/interactiveBrowserCredential.browser.ts

```diff
@@ -147,14 +154,35 @@ export class InteractiveBrowserCredential implements TokenCredential {
   ): Promise<AccessToken | null> {
     const { span } = createSpan("InteractiveBrowserCredential-getToken", options);
     try {
-      if (!this.msalObject.getAccount()) {
-        await this.login();
+      const currentAccounts = this.msalObject.getAllAccounts();
+      if (!currentAccounts || !currentAccounts.length) {
+        const result = await this.login();
+        if (result && result.account) {
+          this.account = result.account;
+        } else {
+          logger.getToken.info("No login response");
+          return null;
+        }
       }
 
       const authResponse = await this.acquireToken({
+        authority: this.msalConfig.auth.authority!,
+        correlationId: this.correlationId,
```

Comment:
I'm currently generating this locally. I don't allow the users to send one through yet, since I haven't checked how to do the same on the Node side of things. Please let me know if this approach looks half decent at least, and what would you like to see to take based on this, or if you'd prefer a different approach.

Reply:
Should be fine. We do this on our end:
```javascript
correlationId = request.correlationId || this.config.cryptoInterface.createNewGuid();
```
Which implies, if you pass the correlationId, we respect it. The only thing I would recommend is to not recycle correlationIds, since they are tied up with a request and show up in server logs.

Reply:
Thanks! I'll send it undefined all the way through if the user doesn't provide one then 👍